storage/country_tree.hpp

```diff
@@ -18,12 +21,14 @@ class CountryTree
   /// \brief m_children contains all first generation descendants of the node.
   /// Note. Once created the order of elements of |m_children| should not be changed.
   /// See implementation of AddAtDepth and Add methods for details.
-  vector<unique_ptr<CountryTree<T>>> m_children;
+  vector<shared_ptr<CountryTree<T>>> m_children;
```

Comment:
Yes, I'll implement the variant with unique_ptr and raw pointers in hash table (unordered_multimap<T, Node*>).